nixos/modules/services/web-apps/pr-tracker.nix

```diff
@@ -0,0 +1,304 @@
+        LimitNOFILE = "1048576";
+        PrivateTmp = true;
+        PrivateDevices = true;
+        StateDirectoryMode = "0700";
+      };
+    in
+    mkIf cfg.enable {
+      users.users = optionalAttrs (cfg.user == "pr-tracker") {
+        pr-tracker = {
+          group = cfg.group;
+          home = cfg.dataDir;
+          isSystemUser = true;
+        };
+      };
+
+      users.groups = optionalAttrs (cfg.group == "pr-tracker") { pr-tracker = { }; };
+
+      networking.firewall = mkIf cfg.openFirewall { allowedTCPPorts = [ 3000 ]; };
+
+      systemd.sockets.pr-tracker = {
+        listenStreams = [ "0.0.0.0:3000" ];
+        wantedBy = [ "sockets.target" ];
+      };
+
+      systemd.services.pr-tracker = {
+        path = [ pkgs.git ];
+
+        serviceConfig =
+          optionalAttrs useClone { ExecStartPre = prestart; }
+          // commonUnitSettings
+          // {
+            Restart = "always";
+
+            StandardInput = "file:${cfg.githubApiTokenFile}";
+
+            ExecStart = concatStringsSep " " (
+              [
+                "${cfg.package}/bin/pr-tracker"
+                "--source-url ${escapeShellArg cfg.sourceUrl}"
+                "--user-agent ${escapeShellArg cfg.userAgent}"
+                "--path ${cfg.nixpkgsClone.cloneDir}"
+                "--remote ${cfg.nixpkgsClone.remote}"
+              ]
+              ++ optionals (cfg.mountPath != null) [ "--mount ${cfg.mountPath}" ]
+            );
+
+            # Hardening
+            CapabilityBoundingSet = "";
+            LockPersonality = true;
+            MemoryDenyWriteExecute = true;
+            PrivateUsers = true;
+            ProtectClock = true;
+            ProtectControlGroups = true;
+            ProtectHome = true;
+            ProtectHostname = true;
+            ProtectKernelLogs = true;
+            ProtectKernelModules = true;
+            ProtectKernelTunables = true;
+            ProtectProc = "invisible";
+            ProcSubset = "pid";
+            ProtectSystem = "strict";
+            RestrictAddressFamilies = [
+              "AF_INET"
+              "AF_INET6"
+              "AF_NETLINK"
+            ];
+            RestrictNamespaces = true;
+            RestrictRealtime = true;
+            SystemCallArchitectures = "native";
+            SystemCallFilter = [
+              "@system-service"
+              "@pkey"
+            ];
+            UMask = "0077";
+          };
+      };
+
+      systemd.timers.pr-tracker-update = optionalAttrs useClone {
+        wantedBy = [ "timers.target" ];
+        timerConfig = {
+          OnBootSec = cfg.nixpkgsClone.interval;
+          OnUnitActiveSec = cfg.nixpkgsClone.interval;
+        };
+      };
+
+      systemd.services.pr-tracker-update = optionalAttrs useClone {
+        path = with pkgs; [
+          git
+          curl
+        ];
+
+        script = ''
+          set -eu
+
+          git -C ${cfg.nixpkgsClone.cloneDir} fetch
+          curl http://localhost:3000/update
```

Comment:
What does this do?

Reply:
oops, this is a remnant of me trying to use this guy's fork https://github.com/PatrickDaG/pr-tracker

I will get rid of it